obal/data/modules/repoclosure.py

```diff
@@ -0,0 +1,55 @@
+"""
+Run repoclosure on a set of repositories
+"""
+
+from subprocess import check_output, CalledProcessError, STDOUT
+from ansible.module_utils.basic import AnsibleModule
+
+
+def main():
+    """
+    Run repoclosure on a set of repositories
+    """
+    module = AnsibleModule(
+        argument_spec=dict(
+            config=dict(type='str', required=True),
+            check=dict(type='list', required=True),
+            additional_repos=dict(type='list', required=False, default=[]),
+            lookaside=dict(type='list', required=False, default=[]),
+        )
+    )
+
+    config = module.params['config']
+    check = module.params['check']
+    additional_repos = module.params['additional_repos']
+    lookaside = module.params['lookaside']
+
+    command = [
+        'dnf',
+        'repoclosure',
+        '--refresh',
+        '--newest',
+        '--config',
+        config
+    ]
+
+    for name in check:
+        command.extend(['--check', name])
+
+    for repo in additional_repos:
+        name, url = repo.split(',')
+        command.extend(['--repofrompath', f"{name},{url}"])
```

Comment:
this implies that we call

```
repoclosure:
  additional_repos:
    - something,https://somewhere
    - else,https://somewhereelse
```

right? we could also make that param a list of dicts, which then will make the call:
```
repoclosure:
  additional_repos
    - name: something
      url: https://…
    - name: else
      url: https://…
```

The later is imho more readable (in the playbook), but I have no idea who/where the input is generated from right now.

Reply:
I was torn between these two worlds. One being the follow the way repoclosure defines additional repositories and make it easier to do so via a command line parameter vs. the clean yaml method of using a dictionary.

Reply:
I think the way `dnf` expects its CLI parameters is an implementation detail that should not leak to the user of the module.
(and you can print whatever is in `command` in the error for reproducing locally).

That said, this is not blocking.